### ISSUE TITLE
Spec-hooks requested

### ISSUE THREAD

**Issue by eladalon1983** [w3c/mediacapture-screen-share#213]
Implementers working on other specs have requested (e.g. on [this issue](https://github.com/WICG/compute-pressure/issues/82)) that we add some spec-hooks. For example, some behavior might be allowed/forbidden if a document holds a video track that refers to an active screen-capture. I think this is a sensible addition (keeping out-of-scope **how** these hooks will be used). I propose:
* A document is called an "actively screen-capturing document" if that document holds a reference to a "live" video track derived from a call to getDisplayMedia. (In the future - also getViewportMedia.)
* A user agent is called "actively screen-capturing" if any of its fully-active documents is an actively screen-capturing document.

Thoughts? @alvestrand? @jan-ivar? @youennf?

**Comment by kenchris** [comment]
Thanks for filing this @eladalon1983

**Comment by eladalon1983** [comment]
Pinging @jan-ivar and @youennf for feedback.

**Comment by eladalon1983** [comment]
> Seems reasonable to me. Do we need "actively screen-captured document" too?

Makes sense to me.

> We already have [[devicesLiveMap]], can they use this?

Let's try and evaluate if using [[devicesLiveMap]] would be sufficiently wieldy. How do we use it to express the following ideas?
1. This document is engaged in screen-capture.
2. This document is engaged being screen-captured.

Do we get passable results, or is it hard to use [[devicesLiveMap]] here?

**Comment by eladalon1983** [comment]
@kenchris, do you guys still need this?
@youennf, unless you can express the [two concepts](https://github.com/w3c/mediacapture-screen-share/issues/213#issuecomment-1092870821) well with `[[devicesLiveMap]]`, I think we should merge a PR for this. I'd like to hear budding consensus before I produce it, though. Wdys?

**Comment by kenchris** [comment]
Yes. We still need this. Maybe this can be done with [[devicesLiveMap]] but as I am not expert in this area I might ignore some corner cases. But if you can show me how to use it correctly, I can live with that

**Comment by dontcallmedom** [comment]
@eladalon1983 as editor, and since this is unlikely to impact implementations directly, I think you should feel empowered to propose a PR in the direction you suggest

**Comment by eladalon1983** [comment]
I would have preferred to hear assenting sounds before producing the PR, but given that this one is simple, I've produced it, and will bring it up in the next editors' meeting.

**Comment by kenchris** [comment]
@eladalon1983 when is this meeting?

**Comment by eladalon1983** [comment]
It's usually on Thursdays, unless cancelled due to competing events.

**Comment by kenchris** [comment]
OK, looking forward to the result of the next meeting then!

**Comment by youennf** [comment]
@kenchris, can you provide some clarifications on what the hooks should be (camera, microphone, screen capture? muted or not?). See https://github.com/w3c/mediacapture-screen-share/pull/224#discussion_r874646060 in particular

**Comment by youennf** [comment]
In that case, we probably want to add a hook in mediacapture-main that would check whether any entry in [[devicesLiveMap]] has a value set to true (per document and/or page).

> even if muted.

If all captures of a given page are muted, it seems CPU pressure could be applied on the corresponding page. I wonder whether you might not just want to know whether a page (or a document) has at least one non-muted live capture track.

**Comment by kenchris** [comment]
Yes, that might work. 

I am not exactly sure what you consider muted, only audio, with no video?

**Comment by youennf** [comment]
> I am not exactly sure what you consider muted, only audio, with no video?

I am referring to https://w3c.github.io/mediacapture-main/#track-muted.
Say for instance the screen gets locked, the UA might want the camera to be muted until the screen is no longer locked for instance.

**Comment by kenchris** [comment]
Got it, your suggestion makes sense then
